app/src/components/GameRoutes.tsx

```diff
@@ -15,14 +25,54 @@ function CurrentPlayerProvider(props: {
   joinCode: string
   children: React.ReactNode
 }) {
-  const { data, loading } = useCurrentPlayerQuery({
+  const [joinGame] = useJoinGameMutation()
+  const [getPlayersResponse] = useCurrentPlayerLazyQuery({
     variables: {
       joinCode: props.joinCode,
       clientUuid: clientUuid()
+    },
+    onCompleted: async data => {
+      console.log("got players response data:", data)
     }
   })
 
-  if (!loading && !data?.players[0]) {
+  const playersResponse = useCurrentPlayerQuery({
+    variables: {
+      joinCode: props.joinCode,
+      clientUuid: clientUuid()
+    }
+  })
+
+  const gamesResponse = useGameByJoinCodeQuery({
```

Comment:
i imagine we only want to make this call if we don't see games in the `playersResponse`, but then I get an error about React hooks getting called conditionally